lib/brew_dash/schema/recipe.ex

```diff
@@ -4,21 +4,28 @@ defmodule BrewDash.Schema.Recipe do
 
   @exportable_attributes [
     :image_url,
-    :name
+    :name,
+    :source,
+    :source_id
   ]
   @derive {Jason.Encoder, only: @exportable_attributes}
 
   schema "recipes" do
     field :image_url, :string
     field :name, :string
+    field :source, :string
+    field :source_id, :string
 
     timestamps()
   end
 
   @doc false
   def changeset(recipe, attrs) do
     recipe
-    |> cast(attrs, [:name, :image_url])
+    |> cast(attrs, [:name, :image_url, :source, :source_id])
```

Comment:
Do we usually want to be able to change the `source` with the changeset? I was thinking that would only be in `changeset_with_source` since it is a special case with the upsert.

```
  def changeset_with_source(recipe, attrs),
    do: recipe |> cast(attrs, [:source, :source_id]) |> changeset(attrs) |> unique_constraint([:source, :source_id])
```

Reply:
yeah, good point! I think I at some point while fighting with ecto was thinking the same.